### PR TITLE
Minimal detain and release functionality

### PR DESCRIPTION
**Pull request by ZackPierce**
Adding detain and release functions to `ISignalCommandMap` and `SignalCommandMap` to accomplish closer parity with the reference event-based `CommandMap` and ease the creation of asynchronously-executing commands. I am aware that various other branches have implemented this functionality in addition to other neat utility enhancements, but I thought it would be valuable to the community if Joel's canonical fork contained at least these basic functions for reference. 

The most contentious decision made in this commit was to use generic `Object`s as acceptable parameters for the detain and release functions, rather than `ISignalCommand` or some new `ICommand` type. This choice was in keeping with the core Robotlegs' intentional sidestepping of strong command typing in favor of supporting any `Object` with an `execute()` method and results in equivalent signatures for these methods in the `ICommandMap` and `ISignalCommandMap` interfaces.

### CODE REVIEW

**Comment by joelhooks** [comment]
I think my only issue here would be that while the implementation is trivial a lot of energy has been put into providing unit tests for this library. It was developed with a test first approach for the most part, and all subsequent functionality/defect squishing have been tested. From the commit comment you probably knew I was gonna go there ;)

Sorry for the delay. I missed it.

Most of the other forks I've seen are adding ways to use named signals and other bits I didn't fully understand. They also don't test the added functionality. If there are other bits that I am missing it'd be cool to get them included.

Cheers

**Comment by ZackPierce** [comment]
No worries, and thanks for the feedback.  Trust me, it gives me great pain to submit a pull request without tests.

I can add some tests, but they'll only work in the AIR or debug player runtimes, due to [a limitation of the System.gc() method](http://livedocs.adobe.com/flash/9.0/ActionScriptLangRefV3/flash/system/System.html#gc%28%29).  Is it okay to add such unit tests, or does that break the general principle of only adding tests that work for all realistic use cases?

In other words, as much as I don't want to leave stuff untested, I also don't want to be the guy that adds tests that fail for everyone not executing tests with a particular runtime version.  (Unless I get your blessing for said sin...)

**Comment by joelhooks** [comment]
You mean so that you can make sure GC works? I'm concerned with the direct behavior than with testing the behavior of Flash Player. Does retaining add this to the Dictionary. Does releasing it remove it. That'd be enough to satisfy my pedantry :>

**Comment by ZackPierce** [comment]
Here they are, unit tests of the detain and release.  Since I didn't want to expose the protected-scope detainedCommands property to external modification or inspection, I ended up making a new test-case-class which extends SignalCommandMap.  This makes the individual unit test Before/After code a bit uglier, essentially functioning as mock-constructor, mock-destructor, but keeps the public API changes small.

**Comment by joelhooks** [comment]
Ya, the two approaches are extend as you've done or extend and override and test that one. This looks fine. Normally I'd bust chops about test naming and whatnot, but it is such a small piece of functionality and it is late ;)
